proposals/NNNN-atf_facade_proposal.md

```diff
@@ -0,0 +1,255 @@
+# ATF Facade
+
+* Proposal: [SDL-NNNN](NNNN-atf_facade_proposal.md)
+* Author: [Oleksandr Deriabin](https://github.com/aderiabin)
+* Status: **Awaiting review**
+* Impacted Platforms: [ATF]
+
+## Introduction
+
+This proposal describes unified interface for ATF (Automated Test Framework) - ATF Facade.
+ATF should provide unified and flexible interface for creation test scripts to perform checks of SDL functionality.
+
+Main ATF Facade purposes:
+ - Simplify access to ATF features for test scripts
+ - Separate test scripts from ATF
+ - Provide well-structured intuitive interface of ATF for test scripts
+ - Unify way of use ATF for creating automated tests
+ - Create single point of ATF features documentation
+ - Provide ability to create more human-readable tests
+
+## Motivation
+
+Currently ATF has a lot of functionality for testing most of SDL features, but each automated test uses different ATF modules.
+ATF does not have unified interface to provide its functionality for test scripts. Instead of this ATF has many separated modules with their own interfaces which are used in automated scripts for SDL testing purposes.
+
+So current approach leads to create test scripts which have multiple direct dependencies on different modules of ATF.
+In other words automated test scripts are fully dependent on ATF implementation and each change in ATF can lead to need for change of test scripts.
+
+In addition current approach requires a strong knowledge in ATF modules purposes and interactions between them for create good automated test scenario using ATF.
+So creation of automated test scripts takes a lot of time and it can be a bit complicated for Mobile application developers in the beginning.
+
+Creation of ATF Facade will allow:
+ - to create test scripts which are independent of ATF implementation
+ - to unify ATF use approach
+ - to document ATF functionality
+ - to improve test scripts readability and their structure on base of scenarious
+
+## Proposed solution
+
+Add new module Facade with next submodules in ATF:
+ - Runner - Manage of run test scenario
+ - SDL - Provide access to manage SDL
+ - HMI - Provide ability to emulate HMI and manage its expectations
+ - Mobile - Provide ability to emulate Mobile device and manage its expectations
+ - Utilities - Provide access to utilities of ATF
+ - Constants - Provide access to all constants used in ATF
```

Comment:
@aderiabin what about config? Expectations?

Reply:
@LuxoftAKutsan Will not be added to base Facade. Can be added in addition.